ISPN-12712 Re-enable DuplicatedDomainsCdiIT

danberindei opened the pull request:
https://issues.redhat.com/browse/ISPN-12712

Remove the cache get operation, since the test is only supposed
to test whether the CDI extension sets the cache manager name or not.

tristantarrant commented:
org.infinispan.test.integration.as.cdi.DuplicatedDomainsCdiIT.testIfCreatingDefaultCacheManagerSucceeds :)

danberindei commented:
This was a weird one... the test deployment imports the Infinispan module in MANIFEST.MF without `meta-inf`, so `InfinispanExtensionEmbedded` does run, but Weld doesn't process any other beans in the extension jar.

To compound the confusion, when I removed the `CdiConfig` class from the deployment, Weld stopped processing the `@Inject` annotation in the test and the test started failing with a diferent NPE. Adding a `beans.xml` resource instead of the `CdiConfig` class re-enabled CDI annotation processing.

The old test didn't include the `CdiConfig` class, so I'm not sure if the WildFly implicit dependency rules changed since we removed the test, or if the old test only passed back then because it didn't try to use the injected cache.

tristantarrant commented:
Merged, thanks
